core/src/main/java/feign/Request.java

```diff
@@ -33,22 +33,37 @@
    */
   public static Request create(String method, String url, Map<String, Collection<String>> headers,
                                byte[] body, Charset charset) {
-    return new Request(method, url, headers, body, charset);
+    return create(method, url, headers, body, charset, false);
+  }
+
+  /**
+   * No parameters can be null except {@code body} and {@code charset}. All parameters must be
+   * effectively immutable, via safe copies, not mutating or otherwise.
+   */
+  public static Request create(String method, String url, Map<String, Collection<String>> headers,
+                               byte[] body, Charset charset, boolean retried) {
+    return new Request(method, url, headers, body, charset, retried);
   }
 
   private final String method;
   private final String url;
   private final Map<String, Collection<String>> headers;
   private final byte[] body;
   private final Charset charset;
+  private final boolean retried;
```

Comment:
was looking at this and also usage in sleuth. Seems a strange place to add state. The template is about how to create a request. The retryer is the place to deal with side-effects about a failed attempt.